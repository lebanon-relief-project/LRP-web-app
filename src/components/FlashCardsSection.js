import React from "react";
import styled from "styled-components";
import colours from "../styles/Colours";
import FlashCard from "./FlashCard";

const flashCard1 = {
  title: "title 1",
  text: "text 1",
  image: "image url",
};
const flashCard2 = {
  title: "title 2",
  text: "text 2",
  image: "image url",
};
const flashCard3 = {
  title: "title 3",
  text: "text 3",
  image: "image url",
};
const flashCard4 = {
  title: "title 4",
  text: "text 4",
  image: "image url",
};
const flashCard5 = {
  title: "title 5",
  text: "text 5",
  image: "image url",
};
const flashCard6 = {
  title: "title 6",
  text: "text 6",
  image: "image url",
};
const flashCard7 = {
  title: "title 7",
  text: "text 7",
  image: "image url",
};
const flashCard8 = {
  title: "title 8",
  text: "text 8",
  image: "image url",
};
const flashCard9 = {
  title: "title 9",
  text: "text 9",
  image: "image url",
};
const flashCard10 = {
  title: "title 10",
  text: "text 10",
  image: "image url",
};
const flashCard11 = {
  title: "title 11",
  text: "text 11",
  image: "image url",
};
const flashCard12 = {
  title: "title 12",
  text: "text 12",
  image: "image url",
};
const flashCard13 = {
  title: "title 13",
  text: "text 13",
  image: "image url",
};
const flashCard14 = {
  title: "title 14",
  text: "text 14",
  image: "image url",
};

const flashCards = [flashCard1, flashCard2, flashCard3, flashCard4, flashCard5, flashCard6, flashCard7, flashCard8, flashCard9,flashCard10,flashCard11,flashCard12,flashCard13,flashCard14];

const FlashCardsSection = (props) => {
  return (
    <StyledSection>
      <Wrapper>
        {flashCards.map((flashCard, index) => {
          return (
            <FlashCard key={`${flashCard.title}_${index}`} card={flashCard} />
          );
        })}
         
      </Wrapper>
      <NextButton>Next</NextButton>

    </StyledSection>
  );
};

const StyledSection = styled.section`
  background: inherit;
  margin: 0;
  padding: 0;
  width: 100%;
  display:flex;
  flex-direction:column;
  justify-content:space-between;
`;

const Wrapper = styled.div`
  flex: 1;
  margin: 0 16.7%;
  background: inherit;
  display: flex;
  flex-wrap: wrap;
  column-gap: 1.4375rem;
  row-gap: 1.4375rem;
`;

const NextButton = styled.button`
  background-color: ${colours.gold};
  width: 13.875rem;
  height: 2.625rem;
  font-size: 1rem;
  align-self: flex-end;
  margin: 3.375rem 16.7%;
  box-shadow: 0px 2px 0px rgba(0, 0, 0, 0.043);
  border-radius: 2px;
  outline: none;
`

export default FlashCardsSection;
