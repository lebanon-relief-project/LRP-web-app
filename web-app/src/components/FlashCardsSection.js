import React, { useEffect, useState } from "react";
import styled from "styled-components";
import colours from "../styles/Colours";
import FlashCard from "./FlashCard";
import { getFlashCards } from "../services/flashCards.service";
import devices from "../styles/Devices";
import { Link } from "react-router-dom";

const FlashCardsSection = (props) => {
  const [flashCards, setFlashCards] = useState([]);
  const fetchFlashCards = async () => {
    let response = await getFlashCards();

    setFlashCards(response.cards);
  };

  useEffect(() => {
    fetchFlashCards();
  }, []);

  return (
    <StyledSection>
      <Wrapper>
        {flashCards.map((flashCard, index) => {
          return (
            <FlashCard
              id={`${flashCard._id}`}
              key={`${flashCard._id}_${index}`}
              card={flashCard}
            />
          );
        })}
      </Wrapper>

  
        <StyledLink to={"/results"}>Give me advice</StyledLink>
 
    </StyledSection>
  );
};

const StyledLink = styled(Link)`
  color: inherit;
  text-decoration: inherit;
  &:focus,
  &:visited,
  &:link {
    text-decoration: none;
    color: inherit;
  }
  padding: 0.5rem;
  justify-content: center;
  display: flex;
  background-color: ${colours.yellow};
  width: 13.875rem;
  height: 2.625rem;
  font-size: 1rem;
  align-self: flex-end;
  margin: 3.375rem 16.7%;
  box-shadow: 0px 2px 0px rgba(0, 0, 0, 0.043);
  border-radius: 2px;
  outline: none;
  border: none;
  &:hover {
    opacity: 0.8;
  }
  &:active {
    opacity: 1;
  }
  @media (max-width: ${devices.mobile}) {
    margin: 3.375rem 0;
    align-self: center;
  }
`;

const StyledSection = styled.section`
  background: inherit;
  margin: 0;
  padding: 0;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
`;

const Wrapper = styled.div`
  width: 70%;
  margin: 0 auto;
  background: inherit;
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  @media (max-width: ${devices.ipadpro}) {
    width: 100%;
    justify-content: center;
  }
`;



export default FlashCardsSection;
