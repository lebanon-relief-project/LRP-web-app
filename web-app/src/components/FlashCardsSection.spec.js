import React from "react";
import { render, wait } from "@testing-library/react";
import FlashCardsSection from "./FlashCardsSection";
import "jest-styled-components";
import { getFlashCards } from "../services/flashCards.service";

jest.mock("./FlashCard", (props) => {
  return (props) => <div>flash card {props.card.title}</div>;
});

jest.mock("../services/flashCards.service", () => {
  return {
    getFlashCards: jest.fn().mockResolvedValue({
      cards: [
        {
          _id: "random _id",
          title: "random title",
          body: "some card body",
        },
      ],
    }),
  };
});

describe("the FlashCardsSection component", () => {
  let container;
  it("should match the snapshot", async () => {
    ({ container } = renderFlashCardsSection());
    await wait();
    expect(container).toMatchSnapshot();
  });
});

const renderFlashCardsSection = () => {
  return render(<FlashCardsSection />);
};
