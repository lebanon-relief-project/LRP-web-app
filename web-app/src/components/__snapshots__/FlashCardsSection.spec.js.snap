// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the FlashCardsSection component should match the snapshot 1`] = `
.c0 {
  background: inherit;
  margin: 0;
  padding: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin: 0 16.7%;
  background: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-column-gap: 1.4375rem;
  column-gap: 1.4375rem;
  row-gap: 1.4375rem;
}

.c2 {
  background-color: #FFD666;
  width: 13.875rem;
  height: 2.625rem;
  font-size: 1rem;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin: 3.375rem 16.7%;
  box-shadow: 0px 2px 0px rgba(0,0,0,0.043);
  border-radius: 2px;
  outline: none;
  border: none;
}

.c2:hover {
  opacity: 0.8;
}

.c2:active {
  opacity: 1;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        flash card 
        random title
      </div>
    </div>
    <button
      class="c2"
    >
      Next
    </button>
  </section>
</div>
`;
